src/lib/services/__tests__/emissionsCalculator.test.ts

```diff
@@ -53,6 +55,57 @@ const withExtendedRanges: IntensityResponseData = new Array(25).fill(null).map((
 })
 
 describe('EmissionsCalculator', function () {
+  const mockCarbon = new MockAgent().get(`https://api.carbonintensity.org.uk`)
+
+  describe('fetchEmissions', function () {
+    const startDateBelow30Min = new Date('2023-01-01T07:00:00.000Z')
+    const endDateBelow30Min = new Date('2023-01-01T07:10:00.000Z')
+
+    this.beforeEach(() => {
```

Comment:
I'd recommend importing `beforeEach` rather than relying on the `this` binding. Can cause weird issues if the `function` gets changed to an `=>` notation

Reply:
Updated, yes, and everything else will be inline also this should've been changed now